File: desktop/src/models/project.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Project {
    pub id: String,
    pub owner_id: String,
    pub name: String,
    pub color: String,
    pub is_archived: bool,
    pub sort_order: i32,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Serialize)]
pub struct CreateProjectPayload {
    pub name: String,
    pub color: String,
    #[serde(rename = "sortOrder")]
    pub sort_order: i32,
}

File: openplan_mobile/shared/src/commonMain/kotlin/com/example/openplan_mobile/data/models/Project.kt
package com.example.openplan_mobile.data.models

import kotlinx.serialization.Serializable

@Serializable
data class Project(
    val id: String,
    val ownerId: String,
    val name: String,
    val color: String,
    val isArchived: Boolean = false,
    val sortOrder: Int = 0,
    val createdAt: String,
    val updatedAt: String
)

@Serializable
data class CreateProjectRequest(
    val name: String,
    val color: String = "#6b7280",
    val sortOrder: Int = 0
)
